sdk/resourcemanager/azure-resourcemanager-eventhubs/src/main/java/com/azure/resourcemanager/eventhubs/implementation/EventHubImpl.java

```diff
@@ -0,0 +1,532 @@
+                        .flatMap(aVoid -> context.voidMono());
+                }));
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureEnabled() {
+            this.ensureSettings().withEnabled(true);
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureDisabled() {
+            this.ensureSettings().withEnabled(false);
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureSkipEmptyArchives(Boolean skipEmptyArchives) {
+            this.ensureSettings().withSkipEmptyArchives(skipEmptyArchives);
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureWindowSizeInSeconds(int sizeInSeconds) {
+            this.ensureSettings().withIntervalInSeconds(sizeInSeconds);
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureWindowSizeInMB(int sizeInMB) {
+            this.ensureSettings().withSizeLimitInBytes(sizeInMB * 1024 * 1024);
+            return this;
+        }
+
+        public CaptureSettings withDataCaptureFileNameFormat(String format) {
+            this.ensureSettings().destination().withArchiveNameFormat(format);
+            return this;
+        }
+
+        public CaptureDescription validateAndGetSettings() {
+            if (this.newSettings == null) {
+                return this.currentSettings;
+            } else if (this.newSettings.destination() == null
+                    || this.newSettings.destination().storageAccountResourceId() == null
+                    || this.newSettings.destination().blobContainer() == null) {
+                throw logger.logExceptionAsError(new IllegalStateException(
+                    "Setting any of the capture properties requires "
+                        + "capture destination [StorageAccount, DataLake] to be specified"));
+            }
+            if (this.newSettings.destination().name() == null) {
+                this.newSettings.destination().withName("EventHubArchive.AzureBlockBlob");
+            }
+            if (this.newSettings.encoding() == null) {
+                this.newSettings.withEncoding(EncodingCaptureDescription.AVRO);
+            }
+            return this.newSettings;
+        }
+
+        private CaptureDescription ensureSettings() {
+            if (this.newSettings != null) {
+                return this.newSettings;
+            } else if (this.currentSettings == null) {
+                this.newSettings = new CaptureDescription().withDestination(new Destination());
+                return this.newSettings;
+            } else {
+                // Clone the current settings to new settings (one time)
+                //
+                this.newSettings = cloneCurrentSettings();
+                return this.newSettings;
+            }
+        }
+
+        private Mono<Void> createContainerIfNotExistsAsync(final StorageAccount storageAccount,
```

Comment:
Mgmt should be able to create a storage blob container?

Reply:
Changed to mgmt storage.